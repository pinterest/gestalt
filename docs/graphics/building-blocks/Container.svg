<svg xmlns="http://www.w3.org/2000/svg" width="312" height="180" fill="none"><path fill="#4A4A4A" fill-rule="evenodd" d="M45 74a1 1 0 0 1 1-1h3.928a1 1 0 1 1 0 2H47v3a1 1 0 1 1-2 0v-4Zm11.786 0a1 1 0 0 1 1-1h7.857a1 1 0 1 1 0 2h-7.857a1 1 0 0 1-1-1ZM72.5 74a1 1 0 0 1 1-1h7.857a1 1 0 1 1 0 2H73.5a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 1 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.715 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 1 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.858a1 1 0 0 1 0 2h-7.858a1 1 0 0 1-1-1Zm15.715 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2H183.5a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 1 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.715 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 1 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1H266a1 1 0 0 1 1 1v4a1 1 0 0 1-2 0v-3h-2.929a1 1 0 0 1-1-1ZM46 85a1 1 0 0 1 1 1v8a1 1 0 1 1-2 0v-8a1 1 0 0 1 1-1Zm220 0a1 1 0 0 1 1 1v8a1 1 0 0 1-2 0v-8a1 1 0 0 1 1-1ZM46 101a1 1 0 0 1 1 1v3h2.929a1 1 0 0 1 0 2H46a1 1 0 0 1-1-1v-4a1 1 0 0 1 1-1Zm220 0a1 1 0 0 1 1 1v4a1 1 0 0 1-1 1h-3.929a1 1 0 0 1 0-2H265v-3a1 1 0 0 1 1-1Zm-209.214 5a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2H73.5a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.715 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 1 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.858a1 1 0 0 1 0 2h-7.858a1 1 0 0 1-1-1Zm15.715 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2H183.5a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 1 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.715 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 0 1 0 2h-7.857a1 1 0 0 1-1-1Zm15.714 0a1 1 0 0 1 1-1h7.857a1 1 0 1 1 0 2h-7.857a1 1 0 0 1-1-1Z" clip-rule="evenodd"/><rect width="100" height="8" x="58" y="86" fill="#767676" fill-opacity=".5" opacity=".8" rx="4"/></svg>