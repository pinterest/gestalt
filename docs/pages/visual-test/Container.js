// @flow strict
import { type Node } from 'react';
import { Box, Container, Text } from 'gestalt';

export default function Snapshot(): Node {
  return (
    <Box color="secondary" display="inlineBlock" padding={1} width={900}>
      <Container>
        <Box color="infoWeak" padding={3}>
          <Text>This is a 800px container</Text>
        </Box>
      </Container>
    </Box>
  );
}
