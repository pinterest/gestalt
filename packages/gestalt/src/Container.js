// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import Box from './Box.js';

type Props = {|
  children?: React.Node,
|};

export default function Container(props: Props) {
  const { children } = props;
  return (
    <Box justifyContent="center" display="flex">
      <Box maxWidth={800} width="100%">
        {children}
      </Box>
    </Box>
  );
}

Container.propTypes = {
  children: PropTypes.node,
};
