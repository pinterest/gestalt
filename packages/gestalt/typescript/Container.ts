import type { Node } from "react";
import Box from "./Box";
type Props = {
  children?: Node;
};
/**
 * https://gestalt.pinterest.systems/Container
 */

export default function Container(props: Props): Node {
  const { children } = props;
  return (
    <Box justifyContent="center" display="flex">
      <Box maxWidth={800} width="100%">
        {children}
      </Box>
    </Box>
  );
}