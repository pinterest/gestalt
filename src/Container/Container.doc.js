// @flow
import * as React from 'react';
import Container from './Container';
import { ns, card, md, PropTable, Example } from '../../docs/src/cards';

ns(
  'Container',
  'Containers are useful in responsively laying out content on different screens.'
);

card(
  <PropTable
    props={[
      {
        name: 'children',
        type: 'any',
      },
    ]}
  />,
  { heading: false }
);

card(
  'Responsive content',
  md`
    On small screens, the container is the width of the screen. On large screens, it centers the content with a max-width of 800px.
  `,
  <Example
    defaultCode={`
<Box color="gray" padding={3}>
  <Container>
    <Box color="white" padding={3}>
      <Text>Centered content</Text>
    </Box>
  </Container>
</Box>
`}
    scope={{ Container }}
    stacked
  />,
  { stacked: true }
);
