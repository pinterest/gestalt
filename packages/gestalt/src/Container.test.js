// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Container from './Container.js';

test('Container renders', () => {
  const component = renderer.create(<Container />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
