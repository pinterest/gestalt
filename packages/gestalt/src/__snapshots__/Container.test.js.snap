// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Renders with children 1`] = `
<div
  className="box justifyCenter xsDisplayFlex"
>
  <div
    className="box"
    style={
      Object {
        "maxWidth": 800,
        "width": "100%",
      }
    }
  >
    <div />
  </div>
</div>
`;

exports[`Container Renders with maxWidth 1`] = `
<div
  className="box justifyCenter xsDisplayFlex"
>
  <div
    className="box"
    style={
      Object {
        "maxWidth": 400,
        "width": "100%",
      }
    }
  >
    <div />
  </div>
</div>
`;

exports[`Container Renders with no props 1`] = `
<div
  className="box justifyCenter xsDisplayFlex"
>
  <div
    className="box"
    style={
      Object {
        "maxWidth": 800,
        "width": "100%",
      }
    }
  />
</div>
`;
